src/super_gradients/training/datasets/dataset_interfaces/dataset_interface.py

```diff
@@ -75,13 +76,18 @@ def __init__(self, dataset_params={}, train_loader=None, val_loader=None, test_l
             self.download_from_cloud()
 
     def download_from_cloud(self):
-        if self.dataset_params.s3_link is not None:
-            env_name = AWS_ENV_NAME
-            downloader = DatasetDataInterface(env=env_name)
-            target_dir = self.dataset_params.dataset_dir
-            if not os.path.exists(target_dir):
-                os.mkdir(target_dir)
-            downloader.load_remote_dataset_file(self.dataset_params.s3_link, target_dir)
+        if self.dataset_params.s3_link is None:
+            return
+
+        # IF WE'RE RUNNING IN DDP, WE SHOULD ONLY DO THIS IN RANK 0
+        with run_once() as should_run:
+            if should_run:
+                env_name = AWS_ENV_NAME
+                downloader = DatasetDataInterface(env=env_name)
+                target_dir = self.dataset_params.dataset_dir
+                if not os.path.exists(target_dir):
+                    os.mkdir(target_dir)
+                downloader.load_remote_dataset_file(self.dataset_params.s3_link, target_dir)
```

Comment:
why aren't we checking if the file itself exist? this will download the same file again even if it exist

Reply:
This is the zip file before It's unpacked. Checking that it exists does not imply the dataset files actually do...Sure there's a solution for this, but Im not sure this PR is the place for it.